### DISCUSSION TITLE
Monitoring with Uptime Kuma

### DISCUSSION THREAD

**Discussion by alexdelli** [wlcrs/huawei_solar#535] [Ideas]
HI, I would like to monitor the connection between HA and Huawei integration  with Uptime Kuma.
How can I set the monitoring?
(ping, https, tcp, mqtt.....) there a lot on function

Thank you

**Comment by Roving-Ronin** [comment]
@alexdelli 

Kuma mentions being able to "Monitoring uptime for HTTP(s) / TCP / HTTP(s) Keyword / HTTP(s) Json Query / Ping / DNS Record / Push / Steam Game Server / Docker Containers" 

You could try pulling the network connection on your inverter(s), then seeing what error message comes up on the HA integration page (i.e. http://10.x.y.z:8123/config/integrations/integration/huawei_solar and then see if you can use the 'HTTP(s) Keyword' function of Kuma to monitor/trigger an alert?

FYI I see there is an integration to pull Kuma monitoring into HA : https://github.com/meichthys/uptime_kuma
